evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -21,6 +21,65 @@
 from evalml.utils import infer_feature_types
 
 
+@pytest.mark.parametrize(
+    "pipeline_class,estimator",
+    [
+        (TimeSeriesRegressionPipeline, "Linear Regressor"),
+        (TimeSeriesBinaryClassificationPipeline, "Logistic Regression Classifier"),
+        (TimeSeriesMulticlassClassificationPipeline, "Logistic Regression Classifier"),
+    ],
+)
+@pytest.mark.parametrize(
+    "thing_thats_wrong", ["not-right-length", "not-separated-by-gap", "both"]
+)
+@pytest.mark.parametrize("gap", [0, 1, 5])
+@pytest.mark.parametrize("forecast_horizon", [1, 5, 10])
+@patch("evalml.pipelines.components.LinearRegressor.fit")
+@patch("evalml.pipelines.components.LogisticRegressionClassifier.fit")
+def test_time_series_pipeline_validates_holdout_data(
+    mock_fit_lr,
+    mock_fit_linear,
+    forecast_horizon,
+    gap,
+    thing_thats_wrong,
+    pipeline_class,
+    estimator,
+    ts_data,
+):
+    pl = pipeline_class(
+        component_graph=[estimator],
+        parameters={
+            "pipeline": {
+                "date_index": None,
+                "gap": gap,
+                "max_delay": 2,
+                "forecast_horizon": forecast_horizon,
+            }
+        },
+    )
+    X, y = ts_data
+    X_train, y_train = X.iloc[:15], y.iloc[:15]
```

Comment:
might be good to just put the 15 into a variable in case we come back to change this for some reason or another down the road.

Reply:
Done!